src/Voron/Impl/Journal/WriteAheadJournal.cs

```diff
@@ -642,8 +642,18 @@ public void ApplyLogsToDataFile(CancellationToken token, TimeSpan timeToWait)
                     _waitForJournalStateUpdateUnderTx.Reset();
                     ExceptionDispatchInfo edi = null;
                     var sp = Stopwatch.StartNew();
+
+                    foreach (var unused in unusedJournals)
```

Comment:
On Dispose, this will probably delete the _journalsToDelete before we updated the header files

Reply:
This is _bad_